packages/react-ui-components/src/Dialog/dialog.tsx

```diff
@@ -0,0 +1,159 @@
+     */
+    readonly isOpen: boolean;
+
+    /**
+     * The handler which gets called once the user clicks on the close symbol in the top right corner of the Dialog.
+     */
+    readonly onRequestClose: () => void;
+
+    /**
+     * The title to be rendered on top of the Dialogs contents.
+     */
+    readonly title: ReactNode;
+
+    /**
+     * The `style` prop defines the visual style of the `Dialog`.
+     */
+    readonly style: DialogStyle;
+
+    /**
+     * The contents to be rendered within the Dialog.
+     */
+    readonly children: ReactNode;
+
+    /**
+     * An Array of nodes(e.g. Action Buttons) which are placed at the bottom of the Dialog.
+     */
+    readonly actions: ReadonlyArray<ReactNode>;
+
+    /**
+     * An optional `className` to attach to the wrapper.
+     */
+    readonly className?: string;
+
+    /**
+     * An optional `contentsClassName` to attach to the content area of the dialog.
+     */
+    readonly contentsClassName?: string;
+
+    /**
+     * An optional css theme to be injected.
+     */
+    readonly theme: DialogTheme;
+}
+
+export class DialogWithoutEscape extends PureComponent<DialogProps> {
+    public render(): JSX.Element {
+        const {
+            className,
+            title,
+            style,
+            children,
+            isOpen,
+            onRequestClose,
+            actions,
+            theme,
+            ...restProps
+        } = this.props;
+
+        const rest = omit(restProps, ['isOpen']); // TODO: ?
+
+        const finalClassName = mergeClassNames(
+            theme.dialog,
+            {
+                [theme['dialog--wide']]: style === 'wide',
+                [theme['dialog--narrow']]: style === 'narrow',
+            },
+            className,
+        );
+
+        const finalClassNameBody = mergeClassNames(
+            theme.dialog__body,
+            'dialog__body'
+        );
+
+        const backDropClassName = mergeClassNames(
+            theme.dialog__backDrop,
+            'dialog__backDrop'
+        );
+
+        return (
+            <Portal isOpened={isOpen}>
+                <section {...rest} className={finalClassName} role="dialog" tabIndex={0}>
+                    <div className={backDropClassName} role="button" onClick={onRequestClose}/>
+                    <div className={theme.dialog__contentsPosition}>
+                        <div className={theme.dialog__contents}>
+                            {onRequestClose && (
+                                <IconButton
+                                    icon="times"
+                                    className={theme.dialog__closeBtn}
+                                    onClick={onRequestClose}
+                                    size="regular"
+                                    style="brand"
+                                    hoverStyle="darken"
+                                    autoFocus={true}
```

Comment:
This automatically focuses the dialog so pressing ESC is actually handled. This is the fix for #2168.